readers/mutation_reader.cc

```diff
@@ -236,18 +277,35 @@ bool mutation_fragment_stream_validating_filter::operator()(mutation_fragment_v2
 }
 
 bool mutation_fragment_stream_validating_filter::operator()(mutation_fragment::kind kind, position_in_partition_view pos) {
-    return (*this)(to_mutation_fragment_kind_v2(kind), pos);
+    return (*this)(to_mutation_fragment_kind_v2(kind), pos, {});
 }
 
 bool mutation_fragment_stream_validating_filter::operator()(const mutation_fragment_v2& mv) {
-    auto valid = (*this)(mv.mutation_fragment_kind(), mv.position());
-    if (valid && mv.is_range_tombstone_change()) {
-        _current_tombstone = mv.as_range_tombstone_change().tombstone();
-    }
-    return valid;
+    return (*this)(mv.mutation_fragment_kind(), mv.position(),
+            mv.is_range_tombstone_change() ? std::optional(mv.as_range_tombstone_change().tombstone()) : std::nullopt);
 }
 bool mutation_fragment_stream_validating_filter::operator()(const mutation_fragment& mv) {
-    return (*this)(to_mutation_fragment_kind_v2(mv.mutation_fragment_kind()), mv.position());
+    return (*this)(to_mutation_fragment_kind_v2(mv.mutation_fragment_kind()), mv.position(), {});
```

Comment:
nit: `std::nullopt` rather than `{}`

Reply:
I think `{}` already conveys the purpose: a parameter whose value we don't care about. We use this in many places in the code.